src/execution/one-shot.ts

```diff
@@ -515,23 +515,30 @@ export class OneShotExecution extends BaseExecution<OneShotScriptConfig> {
   }
 
   /**
-   * Read this script's fingerprint file.
+   * Read this script's previous fingerprint from `fingerprint` file in the
+   * `.wireit` directory. Cached after first call.
    */
   async #readPreviousFingerprint(): Promise<Fingerprint | undefined> {
-    try {
-      return Fingerprint.fromString(
-        (await fs.readFile(
-          this.#fingerprintFilePath,
-          'utf8'
-        )) as FingerprintString
-      );
-    } catch (error) {
-      if ((error as {code?: string}).code === 'ENOENT') {
-        return undefined;
-      }
-      throw error;
+    if (this.#cachedPreviousFingerprint === undefined) {
+      this.#cachedPreviousFingerprint = (async () => {
+        try {
+          return Fingerprint.fromString(
+            (await fs.readFile(
+              this.#fingerprintFilePath,
+              'utf8'
+            )) as FingerprintString
+          );
+        } catch (error) {
+          if ((error as {code?: string}).code === 'ENOENT') {
+            return undefined;
+          }
+          throw error;
+        }
+      })();
     }
+    return this.#cachedPreviousFingerprint;
   }
+  #cachedPreviousFingerprint?: Promise<Fingerprint | undefined>;
```

Comment:
Do we use standard class fields? If not, consider an explicit init to undefined

Reply:
We use down-leveled standard class fields (though maybe we shouldn't given the extra book-keeping, it's because we originally emitted native class fields, but then downgraded to Node 14 support).